security/webauthn/src/test/java/AbstractWebAuthnTest.java

```diff
@@ -0,0 +1,215 @@
+
+    }
+
+    public static String getChallenge(String userName, Filter cookieFilter) {
+        JsonObject registerJson = new JsonObject().put("name", userName);
+        ExtractableResponse<Response> response = given()
+                .body(registerJson.encode())
+                .contentType(ContentType.JSON)
+                .filter(cookieFilter)
+                .post(REGISTER_URL)
+                .then()
+                .statusCode(200)
+                .cookie("_quarkus_webauthn_challenge", Matchers.notNullValue())
+                .cookie("_quarkus_webauthn_username", Matchers.notNullValue()).extract();
+        JsonObject responseJson = new JsonObject(response.asString());
+        String challenge = responseJson.getString("challenge");
+        Assertions.assertNotNull(challenge);
+        return challenge;
+    }
+
+    private void verifyLoggedIn(Filter cookieFilter, String userName, User user) {
+
+        // public API still good
+        given().filter(cookieFilter)
+                .get(PUBLIC_API_URL)
+                .then()
+                .statusCode(200)
+                .body(Matchers.is("public"));
+        // public API user name
+        given().filter(cookieFilter)
+                .get(PUBLIC_ME_API_URL)
+                .then()
+                .statusCode(200)
+                .body(Matchers.is(userName));
+
+        // user API accessible
+        given().filter(cookieFilter)
+                .get(USER_API_URL)
+                .then()
+                .statusCode(200)
+                .body(Matchers.is(userName));
+
+        //admin API
+        if (user == User.ADMIN) {
+            RestAssured.given().filter(cookieFilter)
+                    .when()
+                    .get("/api/admin")
+                    .then()
+                    .statusCode(200)
+                    .body(Matchers.is("admin"));
+        } else {
+            RestAssured.given().filter(cookieFilter)
+                    .when()
+                    .get("/api/admin")
+                    .then()
+                    .statusCode(403);
+        }
+
+    }
+
+    private void verifyLoggedOut(Filter cookieFilter) {
+        // public API still good
+        given().filter(cookieFilter)
+                .get(PUBLIC_API_URL)
+                .then()
+                .statusCode(200)
+                .body(Matchers.is("public"));
+        // public API user name
+        given().filter(cookieFilter)
+                .get(PUBLIC_ME_API_URL)
+                .then()
+                .statusCode(200)
+                .body(Matchers.is("<not logged in>"));
+
+        // user API not accessible
+        given()
+                .filter(cookieFilter)
+                .redirects().follow(false)
+                .get(USER_API_URL)
+                .then()
+                .statusCode(302)
+                .header("Location", Matchers.is(getApp().getHost() + "/login.html"));
```

Comment:
`getApp().getHost()` is deprecated, also using `getApp().getURI(Protocol.HTTP))` solves test failures.